src/Symfony/Component/Notifier/Recipient/PhoneValidationTrait.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Recipient;
+
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+
+/**
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ *
+ * @internal
+ */
+trait PhoneValidationTrait
+{
+    private static function ensurePhoneIsValid(string $phone): void
```

Comment:
Not a big fan of having validation of Email/Phone in Notifier. First, we know first hand that validating an email is complex and people will want to override the default behavior (same for phone numbers, we might have a provider that does not support having an international code for instance). I would remove the validation and let the developer do it before creating Recipient instances.

Reply:
Makes total sense to me. Thanks for the explanation. I will remove the validation.